geode-core/src/main/java/org/apache/geode/internal/security/SecurityService.java

```diff
@@ -45,7 +45,7 @@
 
   void authorize(Resource resource, Operation operation, String target);
 
-  void authorize(Resource resource, Operation operation, String target, String key);
+  void authorize(Resource resource, Operation operation, String target, Object key);
 
   void authorize(Resource resource, Operation operation, Target target, String key);
```

Comment:
Should this one also be changed to have the "key" parameter be Object instead of String?

Reply:
ditto

Reply:
This method is called in all 3 places with key ResourcePermission.ALL which is
public static final String ALL = "*";

Reply:
Good point. I'd say if any change is made to this it should be for a different jira ticket and not part of this pull request.